src/main/java/paperai/pdfextract/PDFExtractor.java

```diff
@@ -44,6 +44,22 @@ public static void main(String[] args) {
         }
     }
 
+    public static String processFileToString(File file) throws IOException {
+        PDDocument doc = PDDocument.load(file);
+
+        ByteArrayOutputStream bos = new ByteArrayOutputStream();
+        BufferedOutputStream buf = new BufferedOutputStream(bos);
+        Writer w = new BufferedWriter(new OutputStreamWriter(bos, "UTF-8"));
```

Comment:
Use a try-with-resource here:

```
try (Writer w = ...) {
 ...
}
```

That closes the write automatically when the block is left.

I believe you can also use `StandardCharsets.UTF_8` here in stead of a string literal.

The buffered stream is not necessary since you are writing to memory anyway. Buffering won't speed up anything.

Reply:
Well I also didn't even reference the buffer I see now...

I will change it.

Reply:
Btw, I believe there is a `StringWriter` which may make this even simpler :)

Reply:
I was too quick, already commited, should I still look into `StringWriter`?

Reply:
Your choice :) I'll approve it. You can still make an update to use the StringWriter before merging. Another approval will not be required.

Reply:
I changed it to Stringwriter